src/System.Reflection.Metadata/src/System/Reflection/Internal/MemoryBlocks/MemoryMappedFileBlock.cs

```diff
@@ -24,22 +25,19 @@ internal unsafe MemoryMappedFileBlock(IDisposable accessor, SafeBuffer safeBuffe
 
         ~MemoryMappedFileBlock()
         {
-            Dispose(false);
+            DisposeImpl();
         }
 
-        protected override void Dispose(bool disposing)
+        public override void Dispose()
         {
-            if (_safeBuffer != null)
-            {
-                _safeBuffer.ReleasePointer();
-                _safeBuffer = null;
-            }
-
-            if (_accessor != null)
-            {
-                _accessor.Dispose();
-                _accessor = null;
-            }
+            DisposeImpl();
+            GC.SuppressFinalize(this);
+        }
+
+        private void DisposeImpl()
+        {
+            Interlocked.Exchange(ref _safeBuffer, null)?.ReleasePointer();
```

Comment:
Since this is SafeBuffer that already has a finalizer, wouldn't it clean up on its own? Actually, it just occurrd to me ... do we even need this object to have a finalizer? The referenced objects do have finalizers or are composed of ones that do (SafeBuffer) so they will be cleaned up automatically if they are not explicitly disposed. Correct?